transform_aes.c

```diff
@@ -141,11 +118,16 @@ static int transop_encode_aes( n2n_trans_op_t * arg,
             encode_uint32( outbuf, &idx, tx_sa_num ); // Not used
 
             /* Generate and encode the IV seed.
-             * Using two calls to rand() because RAND_MAX is usually < 64bit
-             * (e.g. linux) and sometimes < 32bit (e.g. Windows).
+             * Assume rand() to deliver 'RAND_BYTES' random bytes and re-calling rand() 
+             * because RAND_MAX is usually < 64bit (e.g. linux) 
+             * and sometimes < 32bit (e.g. Windows).
              */
-            iv_seed = ((((uint64_t)rand() & 0xFFFFFFFF)) << 32) | rand();
-            encode_buf(outbuf, &idx, &iv_seed, sizeof(iv_seed));
+	    uint32_t random_int;
+	    for (int i=0; i < TRANSOP_AES_IV_SEED_SIZE; i++) {
+		if ((i % RAND_BYTES) == 0) random_int = rand();
+		iv_seed[i] = (random_int >> ((i % RAND_BYTES) * 8)) & 0xFF;
+            }
+            encode_buf(outbuf, &idx, iv_seed, TRANSOP_AES_IV_SEED_SIZE);
```

Comment:
This is not part of this pull request topic, we will discuss this separately, please restore the old code for now

Reply:
D'accord. We should discuss random numbers and their generation separately.